### ISSUE TITLE
[4.x]: Cannot load editor interface on 4.5.10

### ISSUE THREAD

**Issue by joshuabaker** [craftcms/cms#13935]
### Description

The database locks up when loading the entry editor interface on 4.5.10.

The issue is not present in 4.5.9.

### Extra context

Possibly the same underlying issue as #13925

### Craft CMS version

Craft Pro 4.5.10

### PHP version

8.0.13

### Operating system and version

Linux 5.10.76-linuxkit

### Database type and version

MariaDB 10.3.32

### Image driver and version

Imagick 3.5.1 (ImageMagick 6.9.11-60)

### Installed plugins and versions

Plugins were disabled to confirm that these weren’t impacting things.

**Comment by joshuabaker** [comment]
Some extra context, the upgrade was `4.5.7 => 4.5.10`.

**Comment by angrybrad** [comment]
> Possibly the same underlying issue as https://github.com/craftcms/cms/issues/13925

[Unlikely](https://github.com/craftcms/cms/issues/13925#issuecomment-1813164228).

> The database locks up

Can you explain that further? Are you seeing high CPU and memory usage? Low resource usage, but waiting on a query to finish? You run `SHOW FULL PROCESSLIST;` to get a list of the currently running queries.

**Comment by uandco** [comment]
@angrybrad  I can reproduce this too in a non-shared filesystem nginx/php context.

To reproduce:
* Upgrade to 4.5.10
* Access the admin/login page
* View source and note the URL of `login.css`, for example `/cpresources/2ca0edf8/css/login.css?v=1700104854`
* Delete the `cpresources` folders.
* Reload `/cpresources/2ca0edf8/css/login.css?v=1700104854` only, not the `admin/login` page. You get a 404.

In 4.5.9 and lower you can reload the resource URL only and it will regenerate the file.
That doesn't affect all the resources tho, only `login.css`, `login.js` and the custom logo.

You can't reproduce the issue in a setup where both PHP and the http server share the same filesystem as the files are correctly generated when the page is loaded and the httpd will serve them directly.

It looks like the asset bundler "works" but doesn't store the manifest correctly, making PHP think the file should not be regenerated when accessed directly.

**Comment by brandonkelly** [comment]
@uandco Thanks for pointing that out! That was caused by 8840c8a7796d9a5856ad80e4814f1d9fc04c7493, which avoids storing resource source paths in cache except when needed as a fallback, which _should_ have been harmless, except I didn’t realize that the fallback has actually been the only way any of this is working since 4.0.0 due to an order of operations bug. (The source paths are supposed to get stored in the `resourcepaths` DB table on request end—except the code tends to get executed _after_ the `EVENT_AFTER_REQUEST` event has already been triggered.)

Fixed now for the next release via d7b75d6cd41d9d83df6e7fff1d7f9572a49a4b86, which will go out tomorrow.

@joshuabaker It’s unclear whether this is the same issue you’re experiencing. This CP resource path bug has nothing to do with the database. Are you sure that your issue is in fact that the database is locking up?

**Comment by uandco** [comment]
No worries @brandonkelly , thanks for fixing it!

My workaround today was to symlink `cpresources` from `web` to `storage`, which is persistent and shared between my php and nginx containers. I think it's a better approach as files would be served by nginx directly and it should theorically be quicker, but curious to know what your thoughts are.

**Comment by brandonkelly** [comment]
Craft 4.5.11 is out now with that fix. Thanks again @uandco!

Going to close this. @joshuabaker let me know if you’re still having issues in 4.5.11.

**Comment by joshuabaker** [comment]
Thanks, @brandonkelly. 🙏🏻

I think that this is almost certainly our issue too. If there’s something else impacting us I’ll open another issue.
